x-pack/plugins/lens/public/trigger_actions/visualize_field_actions.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import {
+  createAction,
+  ACTION_VISUALIZE_LENS_FIELD,
+  VisualizeFieldContext,
+} from '../../../../../src/plugins/ui_actions/public';
+import { ApplicationStart } from '../../../../../src/core/public';
+
+export const visualizeFieldAction = (application: ApplicationStart) =>
+  createAction<typeof ACTION_VISUALIZE_LENS_FIELD>({
+    type: ACTION_VISUALIZE_LENS_FIELD,
+    id: ACTION_VISUALIZE_LENS_FIELD,
+    getDisplayName: () =>
+      i18n.translate('xpack.lens.discover.visualizeFieldLegend', {
+        defaultMessage: 'Visualize field',
+      }),
+    isCompatible: async () => !!application.capabilities.visualize.show,
```

Comment:
this `capabilities.visualize.show` caught my eye, just confirming that this is correct capability for lens?

Reply:
Yes, we did the same for maps too!